python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,566 @@
+
+            subgraphs_nodes_mapping.append(new_subgraph_nodes[biggest_comp])
+            subgraphs.append(new_subgraph_big_comp)
+
+        return subgraphs, subgraphs_nodes_mapping, pruned_nodes
+
+    def explain_graph(self, graph, feat, **kwargs):
+        r"""Find the subgraph that play a crucial role to explain the prediction made
+        by the GNN for a graph.
+
+        Parameters
+        ----------
+        graph: DGLGraph
+            A homogeneous graph.
+        feat: Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        kwargs: dict
+            Additional arguments passed to the GNN model. Tensors whose
+            first dimension is the number of nodes or edges will be
+            assumed to be node/edge features.
+
+        Returns
+        -------
+        list
+            Return list of nodes that represent the most important subgraph.
+
+        Examples
+        --------
+
+        >>> import dgl.function as fn
+        >>> import torch
+        >>> import torch.nn as nn
+        >>> from dgl.data import GINDataset
+        >>> from dgl.dataloading import GraphDataLoader
+        >>> from dgl.nn import AvgPooling, SubgraphXExplainer
+
+        >>> # Load dataset
+        >>> data = GINDataset('MUTAG', self_loop=True)
+        >>> dataloader = GraphDataLoader(data, batch_size=64, shuffle=True)
+
+        >>> # Define a model
+        >>> class Model(nn.Module):
+        ...     def __init__(self, in_feats, out_feats):
+        ...         super(Model, self).__init__()
+        ...         self.linear = nn.Linear(in_feats, out_feats)
+        ...         self.pool = AvgPooling()
+        ...
+        ...     def forward(self, graph, feat, eweight=None):
+        ...         with graph.local_scope():
+        ...             feat = self.linear(feat)
+        ...             graph.ndata['h'] = feat
+        ...             if eweight is None:
+        ...                 graph.update_all(fn.copy_u('h', 'm'), fn.sum('m', 'h'))
+        ...             else:
+        ...                 graph.edata['w'] = eweight
+        ...                 graph.update_all(fn.u_mul_e('h', 'w', 'm'), fn.sum('m', 'h'))
+        ...             return self.pool(graph, graph.ndata['h'])
+
+        >>> # Train the model
+        >>> feat_size = data[0][0].ndata['attr'].shape[1]
+        >>> model = Model(feat_size, data.gclasses)
+        >>> criterion = nn.CrossEntropyLoss()
+        >>> optimizer = torch.optim.Adam(model.parameters(), lr=1e-2)
+        >>> for bg, labels in dataloader:
+        ...     logits = model(bg, bg.ndata['attr'])
+        ...     loss = criterion(logits, labels)
+        ...     optimizer.zero_grad()
+        ...     loss.backward()
+        ...     optimizer.step()
+
+        >>> # Initialize the explainer
+        >>> explainer = SubgraphXExplainer(model, num_gnn_layers=1, coef=6, high2low=True,
```

Comment:
Use more default values to simplify L481-482?

Reply:
done.